File: tests/07_phases_ompss.dg/fortran/success_task_33.f90
! <testinfo>
! test_generator=config/mercurium-ompss
! test_compile_fail_nanos6_mercurium=yes
! test_compile_fail_nanos6_imfc=yes
! </testinfo>
PROGRAM MAIN
    IMPLICIT NONE
    INTEGER :: LOCAL

    LOCAL = 1
    CALL FOO(1)
    !$OMP TASKWAIT

    LOCAL = 2
    CALL FOO(2)
    !$OMP TASKWAIT
CONTAINS
    !$OMP TASK
    SUBROUTINE FOO(CHECK)
        IMPLICIT NONE
        INTEGER, VALUE :: CHECK

        IF (CHECK /= LOCAL) STOP 1
    END SUBROUTINE
END PROGRAM MAIN

File: tests/07_phases_ompss.dg/cxx/success_task_33.cpp
/*
<testinfo>
test_generator=(config/mercurium-ompss "config/mercurium-ompss-2 openmp-compatibility")
</testinfo>
*/

#pragma omp task inout(*var) node(*var)
void foo(int* var)
{
    (void)var;
}

#pragma omp task inout(*var) stream(*var2)
void foo2(int* var2)
{
    (void)var2;
}


int main()
{
    int a = 1;
    foo(&a);

    foo2(&a);

    #pragma omp task node(0)
    {
    }

	#pragma omp task stream(1)
    {
    }

    #pragma omp taskwait
}
